tensilelite/Tensile/TensileInstructions/Base.py

```diff
@@ -48,36 +50,30 @@ def __new__(cls, *args, **kwargs):
                 cls._instance._kernelInfo = {}
         return cls._instance
 
-    @dataclass
-    class IsaInfo:
-        assemblerPath: str
-        asmCaps: dict
-        archCaps: dict
-        regCaps: dict
-        asmBugs: dict
 
     @dataclass
     class kernelInfo:
-        isa: Tuple[int, int, int]
+        isa: IsaVersion
         wavefrontSize: int = 64
 
-    def init(self, isaVersion: Tuple[int, int, int], assemblerPath: str, debug: bool=False) -> None:
+    def init(self, isaVersion: IsaVersion, assemblerPath: str, debug: bool=False) -> None:
         with self._lock:
             if len(self._kernelInfo) > 1000:
                 self._kernelInfo = _removeIdent(self._kernelInfo)
             self._kernelInfo[threading.get_ident()] = TensileInstructions.kernelInfo(isa=isaVersion)
             if isaVersion not in self._isaInfo: # type: ignore
+                start = timer()
                 asmCaps  = initAsmCaps(isaVersion, assemblerPath, debug)
```

Comment:
Is lru_cache faster then getting the dictionary data from singleton? The original design is to use a singleton to get the device properties anywhere in the generator in one thread. Including wavefronts, and slc, glc support when outputting strings from Item.